cmd/maya-apiserver/app/server/volume_endpoint_v1alpha1.go

```diff
@@ -238,3 +257,35 @@ func (v *volumeAPIOpsV1alpha1) list() (*v1alpha1.CASVolumeList, error) {
 	glog.Infof("cas template based volumes were listed successfully: namespaces '%s'", vols.Namespace)
 	return cvols, nil
 }
+
+func (v *volumeAPIOpsV1alpha1) stats(volName string) (interface{}, error) {
+	glog.Infof("CASTemplate based stats request received")
+
+	cvol, err := v.read(volName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(cvol.Spec.TargetIP) == 0 {
+		return nil, fmt.Errorf("Invalid target IP")
+	}
+
+	targetURL := "http://" + cvol.Spec.TargetIP + exporterPort + metricPath
+
+	mfChan := make(chan *dto.MetricFamily, 1024)
+
+	go func() {
+		err := prometheus.FetchMetricFamilies(targetURL, mfChan, "", "", false)
```

Comment:
Can we just use the http endpoints exposed by the sidecar of volume target to get the stats?
Why should the code be dependent on prometheus libs?

Reply:
we are not using this anymore.